src/Resources/skeleton/authenticator/login_form.tpl.php

```diff
@@ -0,0 +1,31 @@
+{% extends 'base.html.twig' %}
+
+{% block title %}Login!{% endblock %}
+
+{% block body %}
+<form class="form-signin" method="post">
+    {% if error %}
+    <div class="alert alert-danger">{{ error.messageKey|trans(error.messageData, 'security') }}</div>
+    {% endif %}
+
+    <h1 class="h3 mb-3 font-weight-normal">Please sign in</h1>
+    <label for="inputEmail" class="sr-only">Email address</label>
+    <input type="email" value="{{ last_username }}" name="email" id="inputEmail" class="form-control" placeholder="Email address" required autofocus>
+    <label for="inputPassword" class="sr-only">Password</label>
+    <input type="password" name="password" id="inputPassword" class="form-control" placeholder="Password" required>
```

Comment:
As you noticed, we "lightly" try to add Bootstrap4 generated code. The idea is to help people who rare using that, but not get too annoying. I noticed *some* classes on this for Bootstrap? How does this actually look? Are some Bootstrap classes missing? Or does this more or less have what it needs?

Reply:
i took the login template file as you've provided ;)
i think some markup is missing to fill bootstrap needs (e.g.: `form-group`), but if we add these, if will be annoying for users that don't use it.
Then whether we remove all boostrap stuff OR we ask the user if he uses bootstrap, and then we provide all bootstrap classes and markup.
But then, maybe we have to handle bootstrap 3 & boostrap 4...

what do you think ?